src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -226,6 +228,9 @@ private class Project : IDisposable
             internal bool ContainsAnySourceFiles(Func<Uri, bool>? filter = null) =>
                 this.specifiedSourceFiles?.Any(filter ?? (_ => true)) ?? false; // keep this as specified, *not* loaded!
 
+            internal bool IsNotebookProject() =>
+                this.ProjectFile == NotebookProjectUri;
```

Comment:
Seems like it could be a lot of string comparisons, but maybe it's a drop in the bucket

Reply:
I think it's probably fine. (It might even be doing a faster pointer comparison first before it tries a string comparison, since we only have one instance of the URI.)